python/cuspatial/cuspatial/core/binpreds/feature_intersects.py

```diff
@@ -33,22 +111,132 @@ def _preprocess(self, lhs, rhs):
         return super()._preprocess(rhs, lhs)
 
 
+class LineStringPointIntersects(IntersectsPredicateBase):
+    def _preprocess(self, lhs, rhs):
+        """Convert rhs to linestrings."""
+        x = cp.repeat(rhs.points.x, 2)
+        y = cp.repeat(rhs.points.y, 2)
+        xy = cudf.DataFrame({"x": x, "y": y}).interleave_columns()
+        parts = cp.arange((len(lhs) + 1)) * 2
+        geometries = cp.arange(len(lhs) + 1)
+        ls_rhs = cuspatial.GeoSeries.from_linestrings_xy(xy, parts, geometries)
+        return self._compute_predicate(
+            lhs, ls_rhs, PreprocessorResult(lhs, ls_rhs)
+        )
+
+
+class LineStringMultiPointIntersects(IntersectsPredicateBase):
+    def _preprocess(self, lhs, rhs):
+        """Convert rhs to linestrings."""
+        xy = rhs.multipoints.xy
+        parts = rhs.multipoints.geometry_offset
+        geometries = cp.arange(len(lhs) + 1)
+        ls_rhs = cuspatial.GeoSeries.from_linestrings_xy(xy, parts, geometries)
+        return self._compute_predicate(
+            lhs, ls_rhs, PreprocessorResult(lhs, ls_rhs)
+        )
+
+    def _postprocess(self, lhs, rhs, op_result):
+        """When a multipoint intersects with a linestring, all of the
+        points in the intersection need to be tested for equality with
+        the linestring.
+
+        This code uses the result of the intersection to determine
+        which points from the rhs to test for membership in the lhs.
+
+        Every point in the intersection is tested for membership in
+        the lhs. If the point is not in the lhs, the result is set to
+        False. If the point is in the lhs, the result is set to True.
+
+        This iterates on the number of results in the intersection,
+        which is not ideal. This code calls for the existence of a
+        libcuspatial function that can return the points in rhs
+        that are equal to points in the lhs.
+        """
+        match_indices = self._get_match_indices(lhs, op_result)
+        result = super()._postprocess(lhs, rhs, op_result)
+        intersections = op_result.result[1]
+        x_coords = rhs.lines.x
+        y_coords = rhs.lines.y
+        result = _false_series(len(lhs))
+        for idx in range(len(intersections)):
+            if isinstance(intersections[idx], ShapelyLineString):
+                result[match_indices[idx]] = True
+                continue
+            x_match = intersections.points.x[idx] == x_coords
+            if x_match.sum() > 0:
+                y_match = intersections.points.y[idx] == y_coords
+                if (y_match & x_match).sum() > 0:
+                    result[match_indices[idx]] = True
+        return result
+
+
+class LineStringPolygonIntersects(IntersectsPredicateBase):
```

Comment:
Polygon related intersection tests should involve some kind of Point-In-Polygon test... The linestring and the linestrings made from the rings can have 0 intersection, but the linestring can still have intersection with the polygon (contained).

Reply:
I see your point, I had incorrectly assumed this behavior. I'll look at integrating intersects and pip for this now.

Reply:
Made into `NotImplementedPredicates`.